Marlin/src/pins/pins_BIGTREE_SKR_MINI_V1_1.h

```diff
@@ -0,0 +1,193 @@
+/**
+ * Marlin 3D Printer Firmware
+ * Copyright (C) 2016 MarlinFirmware [https://github.com/MarlinFirmware/Marlin]
+ *
+ * Based on Sprinter and grbl.
+ * Copyright (C) 2011 Camiel Gubbels / Erik van der Zalm
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#ifndef TARGET_STM32F1
+  #error "Oops! Select an STM32F1 board in 'Tools > Board.'"
+#endif
+
+#ifndef BOARD_NAME
+  #define BOARD_NAME "BIGTREE SKR mini V1.1"
+#endif
+
+// Ignore temp readings during develpment.
+//#define BOGUS_TEMPERATURE_FAILSAFE_OVERRIDE
+
+//
+// Limit Switches
+//
+#define U_MIN_PIN          -1
+#define V_MIN_PIN          -1
+#define W_MIN_PIN          -1
+#define X_STOP_PIN         PC2
+#define Y_STOP_PIN         PC1
+#define Z_STOP_PIN         PC0
+
+//
+// Steppers
+//
+
+/**
+ * TODO: Currently using same Enable pin for all steppers.
+ */
+
+#define X_STEP_PIN         PC6
+#define X_DIR_PIN          PC7
+#define X_ENABLE_PIN       PB15
+//#ifndef X_CS_PIN
+//  #define X_CS_PIN         PC10
+//#endif
+
+#define Y_STEP_PIN         PB13
+#define Y_DIR_PIN          PB14
+#define Y_ENABLE_PIN       PB12
+//#ifndef Y_CS_PIN
+//  #define Y_CS_PIN         PC11
+//#endif
+
+#define Z_STEP_PIN         PB10
+#define Z_DIR_PIN          PB11
+#define Z_ENABLE_PIN       PB2
+//#ifndef Z_CS_PIN
+//  #define Z_CS_PIN         PC12
+//#endif
+
+#define E0_STEP_PIN        PC5
+#define E0_DIR_PIN         PB0
+#define E0_ENABLE_PIN      PC4
+//#ifndef E0_CS_PIN
+//  #define E0_CS_PIN        PC13
+//#endif
+
+//#define E1_STEP_PIN        PC8
+//#define E1_DIR_PIN         PC9
+//#define E1_ENABLE_PIN      PA8
+
+//#define E2_STEP_PIN        PC10
+//#define E2_DIR_PIN         PC11
+//#define E2_ENABLE_PIN      PA8
```

Comment:
If these E pins are potentially usable, then they can be un-commented. Marlin will only try to use them if `EXTRUDERS > 1`.

Reply:
@thinkyhead not really there is no need for them there is no place for a E2 driver or headers on the board for an additional driver. Here is a cleaned up version of the pin file:  https://github.com/3DSmitty/Marlin_Bigtreetech_SKR_Mini_v1_1/blob/bugfix-2.0.x/Marlin/src/pins/pins_BIGTREE_SKR_MINI_V1.1.h

Reply:
Also the limit switches should be defined like this:

#define X_MIN_PIN          PC2
#define X_MAX_PIN          PA2
#define Y_MIN_PIN          PC1
#define Y_MAX_PIN          PA1
#define Z_MIN_PIN          PC0
#define Z_MAX_PIN          PC3

because there are headers for both + and - on the board. Wether they are all used or not should be determined by Configuration.h not the pins file.

Reply:
Great, thanks! I will put all of the above into a commit. Anything else I missed?

How do you like the Python script layout? I'd like to avoid having boards require their own `env` entry. I wonder if a single platform-centric Python script could look at the value of `MOTHERBOARD` and take different actions based on that….

Reply:
@thinkyhead Thank you. The python script is fine. This is how I did it as well. The name should be board specific as it doesn't relate to all the stm32 boards. This is the same way mks_robin does it as well.

Reply:
Let's hope these birds don't multiply out of control. Ideally it would be best to have just one build option per platform.